scipy/stats/_stats_py.py

```diff
@@ -376,7 +376,7 @@ def hmean(a, axis=0, dtype=None, *, weights=None):
 ModeResult = namedtuple('ModeResult', ('mode', 'count'))
 
 
-def mode(a, axis=0, nan_policy='propagate'):
+def mode(a, axis=0, nan_policy='propagate', *, keepdims=None):
```

Comment:
the `*` here makes `keepdims` keyword only, right? Is this the desired behavior?

Reply:
@Kai-Striega yes for all new parameters which are optional we ask to add it now. It forces the user to be explicit in their code and allows more flexibility on our side for maintenance as we can more things around. Scikit-learn changed their whole API like this and we decided on our side to only do this for new parameters to add little friction.

Reply:
Ok. I must have missed the discussion. Thanks for bringing me up to speed.

Reply:
I must have missed the discussion, too. I wrote my thoughts about it [here](https://github.com/scipy/scipy/pull/15389#discussion_r786147559), though.